torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -320,36 +284,37 @@ def consolidate_state_dict(self, to: int = 0) -> None:
                 if rank == self.rank:
                     # Directly append own optimizer state
                     self._all_state_dicts.append(
-                        _recursive_copy_to_device(self.optim.state_dict(), non_blocking=True, device=torch.device("cpu"),)
+                        _recursive_copy_to_device(self.optim.state_dict(), non_blocking=True, device=torch.device("cpu"))
                     )
                 else:
                     # Receive the optimizer state from the source rank
-                    local_state_dict = _broadcast_object(
-                        empty_messenger,
-                        src_rank=global_rank,
+                    local_state_dict_list = [None]
+                    dist.broadcast_object_list(
```

Comment:
would I be correct if I assume this can be replaced by a gather when NCCL supports that?

Reply:
Yes, once `gather()` is supported, the entire for loop ranging over the ranks will be replaced with a single `gather_object()`. It may look something like:
```
dist.gather_object(self.optim.state_dict(), [None for _ in range(self.world_size)] if self.rank == to else None, dst=to)
```